text/0813-template-layering.md

```diff
@@ -0,0 +1,1001 @@
+2. Eliminating the need for name-based runtime resolutions
+3. Ability to use imported template constructs
+4. Ability to interact with the surrounding JavaScript scope
+
+[rfc-496]: ./0496-handlebars-strict-mode.md
+
+However, one of the drawbacks is that it requires a custom syntax extension
+(`<template>`) and a custom file format (`.gjs/.gts`), which requires a build
+step and other custom tooling.
+
+RFC #779 makes a good case for why this is necessary and preferable to the
+alternatives. We stand by the rationales given in RFC #779, and this RFC does
+not intend to revisit its recommendations. However, there remain cases where
+this drawback is highly undesirable or simply not acceptable:
+
+1. Until we have implemented good editor integration, the developer experience
+   of using `.gjs`/`.gts` could be abysmal (without any syntax highlighting or
+   completion at all, etc). Even after we ship good editor integration for the
+   mainstream editors, this will likely remain the case for a subset of less
+   used editors that we did not or could not prioritize supporting.
+
+2. There may be editors and environments that are impossible for us to support,
+   either because they don't have an extension system at all, or those systems
+   do not expose the capabilities we need. For example, CodeSandbox, CodePen,
+   TypeScript playground, etc.
+
+3. When publishing packages to NPM, it is important and desirable to publish
+   only standard `.js` files that does not require further processing by the
+   consumer (see [v2 Addon Format](./0507-embroider-v2-package-format.md)).
+
+4. There are use cases that prefer or require components that are directly
+   runnable in the browser without a build step, such as bug report templates,
+   runnable code samples, generating components dynamically at runtime, etc.
+
+From here on, we will refer to users and use cases collectively as "standard
+JavaScript environments". Under these circumstances, users would be unable to
+adopt the `<template>` feature, which means they cannot take advantage of the
+numerous benefits unlocked by the feature, such as the strict mode opt-in.
+
+Also, if an addon chose to only provide template constructs (e.g. helpers and
+components) exclusively via importable modules (as opposed to merging them into
+"App Javascript"), then there will be no easy way to access these template
+constructs from those environments.
+
+It is not technically impossible to accomplish these same goals in standard
+JavaScript environments. After all, the `<template>` feature is specified using
+primitives that already exist. Users in standard JavaScript environments can
+just use those primitives directly.
+
+However, that is not an ideal outcome. `<template>` is a user-facing feature,
+and is poised to (or at least well-positioned to) become the main way Ember
+users read, write and reason about components in the next edition of Ember when
+the feature is fully rolled out.
```

Comment:
`<template>` vs `template()`.  What prevents the latter from wide adoption and use in the community?  Do we just expect users to prefer the format in rfc 779?